shared/torngit/base.py

```diff
@@ -50,8 +50,11 @@ class TorngitBaseAdapter(object):
         "xtend",
     )
 
-    def get_client(self):
-        timeout = httpx.Timeout(self._timeouts[1], connect=self._timeouts[0])
+    def get_client(self, timeouts: List[int] = []) -> httpx.AsyncClient:
+        if timeouts:
+            timeout = httpx.Timeout(timeouts[1], connect=timeouts[0])
+        else:
+            timeout = httpx.Timeout(self._timeouts[1], connect=self._timeouts[0])
```

Comment:
Was this call previously defaulting to `[10,30]` for its timeout? JC what is the advantage of switching over to a 3 second timeout in this case?

Reply:
Yeah, its 30s timeout to hear back from the response (10s timeout to connect), so when it flakes, we hang for 30s in our login request and then the timeout is not handled so the client call to us breaks. This change makes it such that we only give GH 3s to respond and it will catch timeouts and handle the exception (default to not a student). This way the login doesn't hang and break, instead it will wait a bit and still login

Reply:
Awesome, thanks for the explanation.